src/handlers/scheduledEventHandler.ts

```diff
@@ -50,17 +50,15 @@ export async function callDiscordNicknameBatchUpdate(env: env) {
 	}
 
 	const data: NicknameUpdateResponseType = await response.json();
-	if (data?.data.totalUsersStatus !== 0 && data?.data.successfulNicknameUpdates === 0) {
-		throw new Error("Error while trying to update users' discord nickname");
+	if (data?.data.unsuccessfulNicknameUpdates === 0) {
+		try {
+			await namespace.put('DISCORD_NICKNAME_UPDATED_TIME', Date.now().toString());
+		} catch (err) {
+			console.error('Error while trying to update the last nickname change timestamp');
+		}
```

Comment:
Can we also track the failed updates based on Id, and store them it another KV, then try again for the failed ones?

Reply:
I think this can be taken up in different issue, while exploring other viable options.

Reply:
Then please create the issues and add the link here

Reply:
- #29